### DISCUSSION TITLE
"wasm api + svg output" may replace katex, for good

### DISCUSSION THREAD

**Discussion by aarmn** [tectonic-typesetting/tectonic#999] [Ideas]
I was planning to create a mini online math book with unique notations that, as far as I know, only I might use. To my knowledge, the best web-based math rendering engine is KaTeX, but it doesn't have the full power of LaTeX. I'm not sure how difficult it would be, but I believe that by using Rust-based APIs of Tectonic and compiling it for WebAssembly with SVG output support, or finding a way to convert PDF to SVG on the client-side (using either WebAssembly or JavaScript), we could transform front-end math representation with a rich ecosystem like real LaTeX, instead of just a portion of it. It may be challenging, and WebAssembly is currently limited in its support, but I believe this would attract a lot of attention to the project, leading to more contributions. Also, WebAssembly will become the future of the web, sooner or later.

**Comment by pkgw** [comment]
Tectonic is certainly designed with this use case in mind!

Tectonic currently has built in support for emitting HTML output, but it requires a lot of support infrastructure that is not yet widely available or easy for people to use. Basically the only public use case is [tt-weave], which is used to generate a digital book called [XeTeX: A Pseudoprogram][xap]. It doesn't involve a lot of display math but in [module 103][103] you can see some basic superscripts and the kerning of the name TeX.

[tt-weave]: https://github.com/tectonic-typesetting/tt-weave/
[xap]: https://stacks.fullyjustified.net/xap/2022.0/#
[103]: https://stacks.fullyjustified.net/xap/2022.0/#103

Also, there is a program [dvisvgm] that can implement the SVG part externally from Tectonic's in-development HTML support. This will be useful for things like web display of TikZ figures.

[dvisvgm]: https://dvisvgm.de/

**Reply by fenjalien** [reply]
I think obsidian (https://obsidian.md/) could benefit hugely from this! It allows you to create notes with markdown files and has its own extra syntax for math blocks. These blocks are processed using MathJax. If they could be processed using tectonic instead, it would be great!

At the moment there is a hacky way of getting latex capablities by adding a visual processor to codeblocks labelled `latex` for example. https://github.com/fenjalien/obsidian-latex-render, https://github.com/artisticat1/obsidian-tikzjax, https://github.com/thequilo/obsidian-tikz-renderer

**Comment by kpym** [comment]
[SwiftLaTeX](https://github.com/SwiftLaTeX/SwiftLaTeX) used to be powered by wasm version of Tectonic, but not anymore, I think.

**Reply by pkgw** [reply]
As a step in this direction, it would be interesting to get a WASM build going in Tectonic's CI infrastructure. I haven't looked into it at all, and maybe it wouldn't even be that hard?
